catalog/harvest/catalog-harvest-api/src/main/java/org/codice/ddf/catalog/harvest/Harvester.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.harvest;
+
+/**
+ * A {@code Harvester} is responsible for fetching remote resources and notifying any registered
+ * {@link Listener}s of create, update and delete events.
+ *
+ * <p>In the event that a {@code Harvester} goes down, it must remember the last state of the
+ * external server it was harvesting from. Upon a {@code Harvester} coming back up, it should only
+ * notify registered listener of new events.
+ */
+public interface Harvester {
+
+  /**
+   * Registers a new {@link Listener} to receive events from the remote server being harvested from.
```

Comment:
✏️ Source instead of server?

Reply:
I think the term "source" is confusing here because we already have https://github.com/codice/ddf/blob/3efc289a995afbddf1170c716233a34ee82eaab8/catalog/core/catalog-core-api/src/main/java/ddf/catalog/source/Source.java. I would like to consider another name, which would mitigate the sources tab issue.